client/web/src/cody/components/ChatUI/ChatUi.tsx

```diff
@@ -102,11 +103,25 @@ export const ChatUI: React.FC<IChatUIProps> = ({ codyChatStore }): JSX.Element =
                 needsEmailVerificationNotice={NeedsEmailVerificationNotice}
                 contextStatusComponent={ScopeSelector}
                 contextStatusComponentProps={scopeSelectorProps}
+                abortMessageInProgressComponent={AbortMessageInProgress}
+                onAbortMessageInProgress={abortMessageInProgress}
             />
         </>
     )
 }
 
+const AbortMessageInProgress: React.FunctionComponent<{ onAbortMessageInProgress: () => void }> = React.memo(
+    function AbortMessageInProgressButton({ onAbortMessageInProgress }) {
+        return (
+            <div className="d-flex justify-content-center w-100 mt-4 mb-2">
+                <Button onClick={onAbortMessageInProgress} variant="secondary" outline={true} size="sm">
+                    Stop generating
+                </Button>
+            </div>
+        )
+    }
+)
```

Comment:
A couple of suggestions:
1. Extract props into a standalone interface to be consistent with other React components in the monorepo.
2. Do not use `React.memo` [if it's not required to fix a known performance issue](https://kentcdodds.com/blog/usememo-and-usecallback).

```suggestion
interface AbortMessageInProgressProps {
  onAbortMessageInProgress: () => void
}

const AbortMessageInProgress: React.FC<AbortMessageInProgressProps> = props => (
            <div className="d-flex justify-content-center w-100 mt-4 mb-2">
                <Button onClick={props.onAbortMessageInProgress} variant="secondary" outline={true} size="sm">
                    Stop generating
                </Button>
            </div>
        )
    }
)
```